framework/src/physics/DiffusionPhysicsBase.C

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DiffusionPhysicsBase.h"
+#include "MatDiffusion.h"
+#include "MoosePreconditioner.h"
+#include "PetscSupport.h"
+#include "MooseEnumItem.h"
+
+InputParameters
+DiffusionPhysicsBase::validParams()
+{
+  InputParameters params = PhysicsBase::validParams();
+  params.addClassDescription("Base class for creating a diffusion equation");
+
+  params.addParam<VariableName>("variable_name", "u", "Variable name for the equation");
+
+  // Diffusivity
+  params.addParam<MaterialPropertyName>("diffusivity_matprop",
+                                        "Material property defining the diffusion coefficient");
+  params.addParam<MooseFunctorName>("diffusivity_functor", "Functor specifying the diffusivity");
+
+  // Source term
+  params.addParam<MooseFunctorName>("source_functor", "Source term in the diffusion problem");
+  params.addParam<Real>("source_coef", 1, "Coefficient multiplying the source");
+
+  // Boundary conditions
+  params.addParam<std::vector<BoundaryName>>("neumann_boundaries",
+                                             "Boundaries on which to apply a diffusive flux");
+  params.addParam<std::vector<BoundaryName>>("dirichlet_boundaries",
+                                             "Boundaries on which to apply a fixed value");
+  params.addParam<std::vector<MooseFunctorName>>(
+      "boundary_fluxes", "Functors to compute the heat flux on each Neumann boundary'");
+  params.addParam<std::vector<MooseFunctorName>>(
+      "boundary_values", "Functors to compute the heat flux on each Dirichlet boundary'");
+  params.addParamNamesToGroup("neumann_boundaries dirichlet_boundaries boundary_fluxes "
+                              "boundary_values",
+                              "Boundary conditions");
+
+  // Preconditioning is implemented so let's use it by default
+  MooseEnum pc_options("default none", "default");
+  params.addParam<MooseEnum>(
+      "preconditioning", pc_options, "Which preconditioning to use for this Physics");
+
```

Comment:
@loganharbour 

this is the second thing. I would have devs define parameters and anything shared in the relevant PhysicsBase class, and derived classes implement the actual discretization using objects

Reply:
It's not something that is impossible in actions, it just has not been done much (at least in NS)